### DISCUSSION TITLE
How to use fields from superclass entity(base entity) when we map child entity

### DISCUSSION THREAD

**Discussion by jovan996** [mapstruct/mapstruct#3084] [Q&A] [answered]
Hi everyone

I have a problem in mapper method when I'm using properties from parent entity (which has some common auditing fields) in child entity.  The code looks like this:

```java
@Getter
@Setter
@MappedSuperclass
@EntityListeners(AuditingEntityListener.class)
public class BaseEntity {

  @CreatedDate
  protected LocalDateTime createdAt;

  @LastModifiedDate
  protected LocalDateTime updatedAt;

}
```

and i have a child entity:

```java
@Getter
@Setter
@Builder
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Entity(name = "patient")
public class PatientEntity extends BaseEntity {

  @Id
  private UUID id;

  @MapsId
  @Exclude
  @JoinColumn(name = "id")
  @OneToOne(fetch = FetchType.LAZY)
  private UserEntity user;

  @Override
  public boolean equals(Object obj) {

    if (this == obj) {
      return true;
    }
    if (!(obj instanceof PatientEntity)) {
      return false;
    }

    return id != null && id.equals(((PatientEntity) obj).getId());
  }
}
```

My mapper is:

```java
@Mapper
public interface PatientMapper {

  
  @Mapping(target = "user", source = "patient")
  @Mapping(target = "updatedAt", ignore = true)
  @Mapping(target = "createdAt", ignore = true)
  PatientEntity toPatientEntity(Patient patient);

}
```

The problem is that mapstruct don't recognize these two fields createdAt and updatedAt(they are red). Actually when I run mvn clean install i get the following errors:
Unknown property "updatedAt" in result type PatientEntity.PatientEntityBuilder. Did you mean "null"?
Unknown property "createdAt" in result type PatientEntity.PatientEntityBuilder. Did you mean "null"?

Can someone please help me to solve this problem?

**Comment by jovan996** [comment, marked as answer]
I managed to solve the problem. I replaced builders with SuperBuilder annotation and also add that annotation on the superclass.

**Reply by filiphr** [reply]
Great that you managed to find a solution to your problem. In general there are some edge cases when things might not work properly with the Lombok `@SuperBuilder`.
